muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/workspace/WorkspaceMaterialController.java

```diff
@@ -743,24 +743,16 @@ public Long getMaterialCountByWorkspaceAndAssignmentType(WorkspaceEntity workspa
   public List<WorkspaceMaterial> listVisibleWorkspaceMaterialsByAssignmentType(WorkspaceEntity workspaceEntity, WorkspaceMaterialAssignmentType assignmentType) {
     final List<WorkspaceNode> folders = listVisibleWorkspaceFolders(workspaceEntity);
     List<WorkspaceMaterial> workspaceMaterials = workspaceMaterialDAO.listByHiddenAndAssignmentTypeAndParents(Boolean.FALSE, assignmentType, folders);
-
     Collections.sort(workspaceMaterials, new Comparator<WorkspaceMaterial>() {
-      
       @Override
       public int compare(WorkspaceMaterial o1, WorkspaceMaterial o2) {
-        int result = getParentIndex(o1.getParent()) - getParentIndex(o2.getParent());
+        int result = o1.getParent().getOrderNumber() - o2.getParent().getOrderNumber();
         if (result == 0) {
           result = o1.getOrderNumber().compareTo(o2.getOrderNumber());
```

Comment:
Yes, magnitude does not matter but since the two materials aren't necessarily under the same folder, the first comparison has to be made on their folders. Don't really see any way to make this simpler.